chatkit-core/src/main/kotlin/com/pusher/chatkit/rooms/api/JoinedRoomApiTypeMapper.kt

```diff
@@ -0,0 +1,42 @@
+package com.pusher.chatkit.rooms.api
+
+import com.pusher.chatkit.rooms.state.JoinedRoomInternalType
+import com.pusher.chatkit.util.DateApiTypeMapper
+
+internal class JoinedRoomApiTypeMapper {
+    fun toRoomInternalType(room: JoinedRoomApiType): JoinedRoomInternalType {
+
+        var lastMessageAt: Long? = null
+        if (room.lastMessageAt != null) {
+            lastMessageAt = DateApiTypeMapper().mapToEpochTime(room.lastMessageAt)
```

Comment:
generally with DI pattern classes should not construct its collaborators but just declare in the constructor them as dependency

Reply:
👍 good point! I've added it to the constructor (and updated the places in the test where we were creating them)